stablehlo/transforms/Passes.td

```diff
@@ -79,6 +79,20 @@ def StablehloRefineShapesPass : Pass<"stablehlo-refine-shapes", "ModuleOp"> {
   }];
 }
 
+def StablehloRefineParametersPass : Pass<"stablehlo-refine-parameters", "ModuleOp"> {
+  let summary = "Refines the parameter shapes of the main function.";
+  let description = [{
+    Modifies the parameters of the main function using the input type signature.
+    Wraps parameters in bitcast_convert ops to keep the rest of the program the
+    same for shape refinement pass.
+  }];
+  let dependentDialects = ["mlir::stablehlo::StablehloDialect"];
+  let options = [
+    ListOption<"refinedShapesOption", "shapes", "std::string",
```

Comment:
It would be ideal if we limit these string arguments specify only the shape (not type) as shape is all we want to specialize. However, parsing wont be a "one-shot" as it is now. 

what do you think?

Reply:
I went through a few ideas, but I think I like where it is now, it's very explicit and difficult to misuse. You must specify exactly the new type signature of the `main` function to be refined.

See the test cases, some aren't tensors `!stablehlo.token`, how would we specify these? Would we skip them? Then the question becomes, do we only specify the dynamic operands? What if we only want to partially refine, how do we skip an operand, etc.

Specifying everything made for an easy contract, and the programmatic API will be to just pass the operand types to the refinement pass

Reply:
Added some details to the pass description